### PR TITLE
feat(plugins) add new fields for Redis ACL configuration [FT-2222]

### PR DESCRIPTION
**Pull request by mikefero**
### Summary
This adds documentation for the new features that were introduced into the rate-limiting plugin by one of our community users @27ascii. This feature was adapted to other OSS (`response-rate-limiting`) and Enterprise (`graphql-rate-limiting-advanced`, `openid-connect`, `proxy-cache-advanced`, and `rate-limiting-advanced`) plugins to complete the feature.

### Reason
Jira: https://konghq.atlassian.net/browse/FT-2170

GH PRs:
- https://github.com/Kong/kong/pull/8213
- https://github.com/Kong/kong/pull/8216
- https://github.com/Kong/kong-ee/pull/2847
- https://github.com/Kong/kong-ee/pull/2849

### Testing
[Pongo](https://github.com/kong/kong-pongo) or [Gojira](https://github.com/Kong/gojira) (along with [Gojira Enterprise](https://github.com/kong/gojira-enterprise)) can be used.

**Note**: Gojira and Gojira Enterprise required changes to be able to test correctly using keycloak and various Redis versions and configurations for validating OpenID Connect and ACL configuration due to fixed Redis version; there is currently [one open PR](https://github.com/Kong/gojira/pull/36) on Gojira (OSS) that needs to be merged in before testing (or ensure you are using `feat/secure-redis-and-allow-version-change` branch when using Gojira). Update your Gojira Enterprise to properly initialize keycloak.

```
REDIS_VERSION=6 gojira keycloack up
REDIS_VERSION=6 gojira keycloack shell
```

### CODE REVIEW

File: app/_hub/kong-inc/openid-connect/index.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ description: |
   - Signed [JWT][jwt] access tokens ([JWS][jws])
   - Opaque access tokens
   - Refresh tokens
-  - Authorization code
```
Comment (mikefero) [inline app/_hub/kong-inc/openid-connect/index.md:17]: @lena-larionova There were various changes like this where space was trimmed; please let me know if you would like me to isolate in a separate commit or revert these entirely.

File: app/_hub/kong-inc/openid-connect/index.md [hunk 2]
```diff
@@ -1074,12 +1074,33 @@ params:
       default: 6379
       datatype: integer
       description: The Redis port.
+    - name: session_redis_username
+      required: false
+      default: null
+      datatype: string
+      description: |
+        Username to use for Redis connection when the `redis` session storage is defined and ACL authentication is desired.
+        If undefined, ACL authentication will not be performed. This requires Redis v6.0.0+
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/index.md:1083]:
```suggestion
        If undefined, ACL authentication will not be performed. This requires Redis v6.0.0+.
```

File: app/_hub/kong-inc/openid-connect/index.md [hunk 3]
```diff
@@ -1074,12 +1074,33 @@ params:
...
       encrypted: true
       datatype: string
-      description: The Redis password.
+      description: |
+        Password to use for Redis connection when the `redis` session storage is defined.
+        If undefined, no AUTH commands are sent to Redis.
+        This field is deprecated and replaced with `session_redis_password`. The`session_redis_auth` field will
+        continue to work in a backwards compatible way, but it is recommended to use the replacement field.
+        A deprecation warning will be logged in the log file, stating the field's deprecation and planned
+        removal in v3.x.x.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/openid-connect/index.md:1103]:
```suggestion

       {:.important}
        > This field is deprecated and replaced with `session_redis_password`. The`session_redis_auth` 
        field will continue to work in a backwards compatible way, but it is recommended to use the 
        replacement field.
        A deprecation warning will be logged in the log file, stating the field's deprecation and planned
        removal in v3.x.x.
```
Adding a class to make this notice more obvious.

File: app/_hub/kong-inc/proxy-cache-advanced/index.md [hunk 4]
```diff
@@ -197,6 +198,22 @@ params:
...
       description: |
         Sentinel master to use for Redis connection when the `redis` strategy is defined. Defining this value implies using Redis Sentinel.
+    - name: redis.sentinel_username
+      required: semi
+      default:
+      value_in_examples:
+      datatype: string
+      description: |
+            Sentinel username to authenticate with a Redis Sentinel instance.
+            If undefined, ACL authentication will not be performed. This requires Redis v6.2.0+
```
Comment (lena-larionova) [inline app/_hub/kong-inc/proxy-cache-advanced/index.md:208]:
```suggestion
            If undefined, ACL authentication will not be performed. This requires Redis v6.2.0+.
```

File: app/_hub/kong-inc/rate-limiting-advanced/index.md [hunk 5]
```diff
@@ -203,13 +203,22 @@ params:
...
       description: |
         Read timeout to use for Redis connection when the `redis` strategy is defined.
+    - name: redis.username
+      required: semi
+      default:
+      value_in_examples:
+      datatype: string
+      description: |
+        Username to use for Redis connection when the `redis` strategy is defined and ACL authentication is desired.
+        If undefined, ACL authentication will not be performed. This requires Redis v6.0.0+
```
Comment (lena-larionova) [inline app/_hub/kong-inc/rate-limiting-advanced/index.md:213]:
```suggestion
        If undefined, ACL authentication will not be performed. This requires Redis v6.0.0+.
```

File: app/_hub/kong-inc/rate-limiting-advanced/index.md [hunk 6]
```diff
@@ -225,15 +234,22 @@ params:
...
         Sentinel master to use for Redis connections when the `redis` strategy is defined.
         Defining this value implies using Redis Sentinel.
+    - name: redis.sentinel_username
+      required: semi
+      default:
+      value_in_examples:
+      datatype: string
+      description: |
+            Sentinel username to authenticate with a Redis Sentinel instance.
+            If undefined, ACL authentication will not be performed. This requires Redis v6.2.0+
```
Comment (lena-larionova) [inline app/_hub/kong-inc/rate-limiting-advanced/index.md:244]:
```suggestion
            If undefined, ACL authentication will not be performed. This requires Redis v6.2.0+.
```

**Review by lena-larionova — COMMENTED** [review]
Minor edits for missing periods (I know you left them off on purpose, but in the output it really does look odd, since it's a sentence ending. I don't think it causes confusion to add those periods)., and a formatting edit.

Question: Are all of these plugins getting a minor/major version bump to account for these changes? If so, we actually need to make copies of the plugin files pre-edit, and name them the old version number for each one that's getting a bump. 

I can do this part, since it needs a metadata update as well, if version bumps are needed (and I'm used to dealing with this mess)

(And yes, this is a super inefficient way of doing these plugin docs. We've got plans to change that this year.)

**Comment by aboudreault** [comment]
@lena-larionova Feel free to merge this one when you are ready.

**Review by lena-larionova — APPROVED** [review]

**Pull request merged**